tests/Phinx/Db/Adapter/MysqlAdapterTest.php

```diff
@@ -676,73 +681,39 @@ public function testAddColumnWithDefaultLiteral()
-    {
-        $table = new \Phinx\Db\Table('table1', [], $this->adapter);
-        $table->save();
-        $this->assertFalse($table->hasColumn('user_id'));
-        $table->addColumn('user_id', 'integer', ['signed' => false])
-              ->save();
-        $rows = $this->adapter->fetchAll('SHOW COLUMNS FROM table1');
-        $this->assertEquals('int(11) unsigned', $rows[1]['Type']);
-    }
-
-    public function testAddSmallIntegerColumnWithDefaultSigned()
-    {
-        $table = new \Phinx\Db\Table('table1', [], $this->adapter);
-        $table->save();
-        $this->assertFalse($table->hasColumn('user_id'));
-        $table->addColumn('user_id', 'smallinteger')
-              ->save();
-        $rows = $this->adapter->fetchAll('SHOW COLUMNS FROM table1');
-        $this->assertEquals('smallint(6)', $rows[1]['Type']);
-    }
-
-    public function testAddSmallIntegerColumnWithSignedEqualsFalse()
-    {
-        $table = new \Phinx\Db\Table('table1', [], $this->adapter);
-        $table->save();
-        $this->assertFalse($table->hasColumn('user_id'));
-        $table->addColumn('user_id', 'smallinteger', ['signed' => false])
-              ->save();
-        $rows = $this->adapter->fetchAll('SHOW COLUMNS FROM table1');
-        $this->assertEquals('smallint(6) unsigned', $rows[1]['Type']);
+        return [
+            ['integer', [], 'int', '11', ''],
+            ['integer', ['signed' => false], 'int', '11', ' unsigned'],
+            ['smallinteger', [], 'smallint', '6', ''],
+            ['smallinteger', ['signed' => false], 'smallint', '6', ' unsigned'],
+            ['biginteger', [], 'bigint', '20', ''],
+            ['biginteger', ['signed' => false], 'bigint', '20', ' unsigned'],
+        ];
     }
 
-    public function testAddBigIntegerColumnWithDefaultSigned()
+    /**
+     * @dataProvider integerDataProvider
+     */
+    public function testIntegerColumnTypes($phinx_type, $options, $sql_type, $width, $extra)
     {
         $table = new \Phinx\Db\Table('table1', [], $this->adapter);
         $table->save();
         $this->assertFalse($table->hasColumn('user_id'));
-        $table->addColumn('user_id', 'biginteger')
+        $table->addColumn('user_id', $phinx_type, $options)
               ->save();
         $rows = $this->adapter->fetchAll('SHOW COLUMNS FROM table1');
-        $this->assertEquals('bigint(20)', $rows[1]['Type']);
-    }
 
-    public function testAddBigIntegerColumnWithSignedEqualsFalse()
-    {
-        $table = new \Phinx\Db\Table('table1', [], $this->adapter);
-        $table->save();
-        $this->assertFalse($table->hasColumn('user_id'));
-        $table->addColumn('user_id', 'biginteger', ['signed' => false])
-              ->save();
-        $rows = $this->adapter->fetchAll('SHOW COLUMNS FROM table1');
-        $this->assertEquals('bigint(20) unsigned', $rows[1]['Type']);
+        $type = $sql_type;
+        if (!$this->usingMysql8()) {
```

Comment:
This should probably be changed to an explicit feature/requirement support check.

Reply:
Are you asking me to split this into two separate functions (one for mysql 8 and for not)? Or are you saying to block using widths for MySQL 8?

Reply:
Typically, you would want to check for the feature you want to use/avoid not a version number.

Reply:
Unit tests are ok to have special version number checks -- but then you'd simply do the version compare in the unit test.

Reply:
Again, what are you asking for with this review? It's not clear to me.

Let's take the `testAddBooleanColumnWithSignedEqualsFalse` function.

It's currently written as:
```php
    public function testAddBooleanColumnWithSignedEqualsFalse()
    {
        $table = new \Phinx\Db\Table('table1', [], $this->adapter);
        $table->save();
        $this->assertFalse($table->hasColumn('test_boolean'));
        $table->addColumn('test_boolean', 'boolean', ['signed' => false])
              ->save();
        $rows = $this->adapter->fetchAll('SHOW COLUMNS FROM table1');

        $type = $this->usingMysql8() ? 'tinyint' : 'tinyint(1)';
        $this->assertEquals($type . ' unsigned', $rows[1]['Type']);
    }
```

I could split this into two different tests depending on version:
```php
    public function testAddBooleanColumnWithSignedEqualsFalse()
    {
        $this->adapter->connect();
        if ($this->usingMysql8()) {
            $this->markTestSkipped('Need to be using mysql < 8');
        }        
        $table = new \Phinx\Db\Table('table1', [], $this->adapter);
        $table->save();
        $this->assertFalse($table->hasColumn('test_boolean'));
        $table->addColumn('test_boolean', 'boolean', ['signed' => false])
              ->save();
        $rows = $this->adapter->fetchAll('SHOW COLUMNS FROM table1');

        $this->assertEquals('tinyint(1) unsigned', $rows[1]['Type']);
    }

    public function testAddBooleanColumnWithSignedEqualsFalseForMysql8()
    {
        $this->adapter->connect();
        if (!$this->usingMysql8()) {
            $this->markTestSkipped('Need to be using mysql 8+');
        }
        $table = new \Phinx\Db\Table('table1', [], $this->adapter);
        $table->save();
        $this->assertFalse($table->hasColumn('test_boolean'));
        $table->addColumn('test_boolean', 'boolean', ['signed' => false])
              ->save();
        $rows = $this->adapter->fetchAll('SHOW COLUMNS FROM table1');

        $this->assertEquals('tinyint unsigned', $rows[1]['Type']);
    }
```

Version check though is kind of the best way to check for a specific feature given it's just an attribute look-up on the connection that PDO readily exposes, instead of having to run 2+ throwaway queries to determine whether it exists or not.

Reply:
Following the precedent in cakephp, I'm saying we wouldn't have an `isMysql8` helper directly in the driver. We would expose the version number and the unit test can perform any special version-specific hacks we want.

The driver would typically expose helpers like `supportsIntegerWidths()` -- if needed. But, since this is only for unit tests, I think exposing just the version number is good enough since we don't have features bound to that.

Reply:
To be specific, I would replace the `usingMysql8()` helper with `getVersion()`.

Reply:
This is good for now. I misread that it was entirely in the unit test although we might want to put the version wrapper in the adapter.